File: capstone2/src/main/java/kr/mjc/yoonsungbae/capstone2/FirstScreen.java
package kr.mjc.yoonsungbae.capstone2;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class FirstScreen extends JFrame {
    public FirstScreen() {
        setTitle("초, 중, 고급 선택");
        setSize(400, 300);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        JRadioButton beginnerButton = new JRadioButton("초급");
        JRadioButton intermediateButton = new JRadioButton("중급");
        JRadioButton advancedButton = new JRadioButton("고급");

        Font font = new Font("맑은 고딕", Font.PLAIN, 20);
        beginnerButton.setFont(font);
        intermediateButton.setFont(font);
        advancedButton.setFont(font);

        ButtonGroup group = new ButtonGroup();
        group.add(beginnerButton);
        group.add(intermediateButton);
        group.add(advancedButton);

        JPanel buttonPanel = new JPanel();
        buttonPanel.setLayout(new BoxLayout(buttonPanel, BoxLayout.Y_AXIS));
        buttonPanel.setAlignmentX(Component.CENTER_ALIGNMENT);

        buttonPanel.add(Box.createVerticalGlue());
        buttonPanel.add(beginnerButton);
        buttonPanel.add(Box.createVerticalStrut(20));
        buttonPanel.add(intermediateButton);
        buttonPanel.add(Box.createVerticalStrut(20));
        buttonPanel.add(advancedButton);
        buttonPanel.add(Box.createVerticalGlue());

        JButton confirmButton = new JButton("확인");
        confirmButton.setFont(font);
        confirmButton.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                String selectedLevel = null;
                if (beginnerButton.isSelected()) {
                    selectedLevel = "beginner";
                } else if (intermediateButton.isSelected()) {
                    selectedLevel = "intermediate";
                } else if (advancedButton.isSelected()) {
                    selectedLevel = "advanced";
                }

                if (selectedLevel != null) {
                    saveLevelAndProceed(selectedLevel);
                } else {
                    JOptionPane.showMessageDialog(null, "레벨을 선택해주세요.");
                }
            }
        });

        JPanel panel = new JPanel();
        panel.setLayout(new BorderLayout());
        panel.add(buttonPanel, BorderLayout.CENTER);
        panel.add(confirmButton, BorderLayout.SOUTH);

        add(panel);
        setVisible(true);
    }

    private void saveLevelAndProceed(String level) {
        UserPreferences.setLevel(level);
        new ThirdScreen();
        this.dispose();
    }
}

File: capstone/src/main/kotlin/FirstScreen.kt
package yoonsungbae

import javax.swing.*

fun createFirstScreen(): JPanel {
    val panel = JPanel()
    panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
    panel.border = BorderFactory.createEmptyBorder(20, 20, 20, 20)

    val label = JLabel("한 번에 턱걸이를 몇 개까지 할 수 있으세요?")
    label.font = label.font.deriveFont(18.0f)
    panel.add(label)

    val beginner = JRadioButton("초급: 10개 할 수 있다.")
    val intermediate = JRadioButton("중급: 20개 할 수 있다.")
    val advanced = JRadioButton("고급: 30개 이상 할 수 있다.")
    val group = ButtonGroup()
    group.add(beginner)
    group.add(intermediate)
    group.add(advanced)
    panel.add(beginner)
    panel.add(intermediate)
    panel.add(advanced)

    val nextButton = JButton("다음")
    nextButton.addActionListener {
        JOptionPane.showMessageDialog(panel, "선택한 수준: ${group.selection?.actionCommand}")
    }
    panel.add(nextButton)

    return panel
}
